packages/console/src/VncConsole/VncConsole.js

```diff
@@ -1,11 +1,180 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 
-const propTypes = {};
-const defaultProps = {};
+import * as NovncLog from '@novnc/novnc/core/util/logging';
+import RFB from '@novnc/novnc/core/rfb';
 
-const VncConsole = () => <div>VncConsole</div>;
+import { noop } from 'patternfly-react';
 
-VncConsole.propTypes = propTypes;
-VncConsole.defaultProps = defaultProps;
+import VncActions from './VncActions';
+
+const CONNECTING = 'connecting';
+const CONNECTED = 'connected';
+const DISCONNECTED = 'disconnected';
+
+/* eslint no-console: ["warn", { allow: ["error"] }] */
+
+class VncConsole extends React.Component {
+  state = { status: CONNECTING };
+
+  componentDidMount() {
+    const {
+      host,
+      port,
+      path,
+      encrypt,
+      resizeSession,
+      viewOnly,
+      shared,
+      credentials,
+      repeaterID,
+      vncLogging,
+      onInitFailed
+    } = this.props;
+
+    NovncLog.init_logging(vncLogging);
+    try {
+      const protocol = encrypt ? 'wss' : 'ws';
+      const url = `${protocol}://${host}:${port}/${path}`;
+
+      const options = {
+        repeaterID,
+        shared,
+        credentials
+      };
+
+      this.rfb = new RFB(this.novncElem, url, options);
+      this.rfb.addEventListener('connect', this.onConnected);
+      this.rfb.addEventListener('disconnect', this.onDisconnected);
+      this.rfb.addEventListener('securityfailure', this.onSecurityFailure);
+      this.rfb.viewOnly = viewOnly;
+      this.rfb.scaleViewport = false; // if the remote session is smaller than HTML container, the view will be centered
+      this.rfb.resizeSession = resizeSession;
+    } catch (e) {
+      onInitFailed && onInitFailed(e);
+      this.rfb = undefined;
+    }
+  }
+
+  onConnected = () => {
+    this.setState({ status: CONNECTED });
+  };
+
+  onCtrlAltDel = e => {
+    if (this.rfb) {
+      this.rfb.sendCtrlAltDel();
+      this.focusVnc(e);
+    }
+  };
+
+  onDisconnected = e => {
+    this.setState({ status: DISCONNECTED });
+    this.props.onDisconnected(e);
+  };
+
+  onSecurityFailure = e => {
+    this.setState({ status: DISCONNECTED });
+    this.props.onSecurityFailure(e);
```

Comment:
maybe I'm missing it, but do you capture the failure text somewhere? e.g. like at https://github.com/theforeman/foreman/blob/develop/app/assets/javascripts/noVNC.js#L27 ?

Reply:
Not recently, this goes up.

Reply:
Any idea how to handle the textual information within the component? I18n needs to be considered...